fairscale/experimental/nn/ssd_offload.py

```diff
@@ -104,6 +119,7 @@ def __init__(self, shape: Tuple[int, ...], dtype: torch.dtype, requires_grad: bo
         # valid if loaded to memory
         self.tensor: Optional[torch.Tensor] = None
         self.requires_grad = requires_grad
+        self.storage_state = StorageState.UNALLOCATED
 
     @classmethod
     def from_file(
```

Comment:
late again, but would it be valuable to pass in an optional cuda Stream here, so that you can make sure that some of these transfers are done in parallel with some other compute ? I do realize that the rest of the class is not cuda specific though

Reply:
We use the same streams as that used by FSDP so in a sense we are not modifying the CPU offload mechanism that we have in place. This class is only responsible for moving params from disk to CPU memory not the GPU memory. One thing to consider is that we can't use pin_memory when using ssd offload. This is because we would essentially need to account for the entire model memory which leads to OOM errors.